packages/oauth/src/providers/keycloak.ts

```diff
@@ -0,0 +1,212 @@
+import {
+	OAuth2ProviderAuthWithPKCE,
+	createOAuth2AuthorizationUrlWithPKCE,
+	validateOAuth2AuthorizationCode
+} from "../core/oauth2.js";
+import { ProviderUserAuth } from "../core/provider.js";
+import { handleRequest, authorizationHeader } from "../utils/request.js";
+
+import type { Auth } from "lucia";
+
+type Config = {
```

Comment:
I think it'd make more send if you just need to provide the `domain` and `client` rather than all the endpoint urls?

Reply:
I changed and commited. Now domain and realm have to be set. Domain has to be in "domain.com" format.